Sources/SwiftyXPC/XPCFileDescriptor.swift

```diff
@@ -6,17 +6,61 @@
 //
 
 import Darwin
+import System
 
-/// A wrapper around a file descriptor, which can be embedded within XPC messages to send file descriptors to other processes.
+/// A owned wrapper around a file descriptor, which can be embedded within XPC messages to send file descriptors to other processes.
 ///
-/// Included for compatibility with macOS 10.15.
-/// On versions of macOS greater than 11.0, this class is unnecessary, and you can simply use a `FileDescriptor` from the `System` module instead.
+/// This wrapper takes ownership of the file descriptor so user doesn't need to close it manually.
+///
+/// This wrapper serves two purposes:
+/// 1. To provide compatibility with macOS 10.15, which does not support the `FileDescriptor` structure.
+/// 2. To ensure that the file descriptor is closed when the message is embedded.
+///
+/// On versions of macOS greater than 11.0, you can also simply use a `FileDescriptor` from the `System` module. However, the `FileDescriptor`
+/// is not owned, so you will need to make sure the file descriptor is not closed in the sender process before the message is embedded
+/// in an XPC message.
```

Comment:
Do you think that we should be `dup`ing the file descriptor on the way in, too? As the class is currently, we currently just copy the file descriptor into our internal state, so it's still just as possible for the caller to close the descriptor on us after creating this wrapper.

Reply:
Alternatively, we could just put a note in the docs saying that it is considered an error for the caller to use the file descriptor anymore after passing it to this class.

Reply:
That's my first version of the implementation since that feels more secure. 

It's just a little more tedious to use since we need to close the fd immediately most of the time.

I don't have a strong preference here. You can make the final call.

Reply:
Yeah, that's a good point. Let's leave it as it is for now; I can consider future tweaks to it down the line.